src/fetch-map/source.ts

```diff
@@ -0,0 +1,223 @@
+    spatialDataColumn,
+    ...(columns && {columns}),
+    ...(filters && {filters}),
+    ...(aggregationExp && {aggregationExp}),
+  } as VectorTableSourceOptions;
+
+  if (type === 'raster') {
+    return rasterSource({
+      ...sourceOptions,
+      tableName: source,
+      ...(filters && {filters: filters as any}),
+    });
+  }
+  if (type === 'tileset') {
+    return vectorTilesetSource({...tilesetOptions});
+  }
+
+  if (type === 'table') {
+    if (isH3) {
+      return h3TableSource({
+        ...(tableOptions as H3TableSourceOptions),
+        ...spatialIndexOptions,
+      });
+    } else if (isQuadbin) {
+      return quadbinTableSource({
+        ...(tableOptions as QuadbinTableSourceOptions),
+        ...spatialIndexOptions,
+      });
+    } else {
+      return vectorTableSource({
+        ...(tableOptions as VectorTableSourceOptions),
+        ...vectorOptions,
+      });
+    }
+  }
+
+  if (type === 'query') {
+    if (isH3) {
+      return h3QuerySource({
+        ...(queryOptions as H3QuerySourceOptions),
+        ...spatialIndexOptions,
+      });
+    } else if (isQuadbin) {
+      return quadbinQuerySource({
+        ...(queryOptions as QuadbinQuerySourceOptions),
+        ...spatialIndexOptions,
+      });
+    } else {
+      return vectorQuerySource({
+        ...(queryOptions as VectorQuerySourceOptions),
+        ...vectorOptions,
+      });
+    }
+  }
+  throw new Error(`Invalid source type: ${type}`);
+}
+
+function getSourceOptions({
+  accessToken,
+  apiBaseUrl,
+  connection,
+  headers,
+  maxLengthURL,
+}: FetchDatasetOptions) {
+  return {
+    accessToken,
+    connectionName: connection,
+    apiBaseUrl,
+    headers,
+    maxLengthURL,
+    ...(headers?.['Cache-Control']?.includes('no-cache') && {
+      localCache: {
+        cacheControl: ['no-cache'] as 'no-cache'[],
+      },
+    }),
+  };
+}
+
+function getDynamicTileResolution(
+  spatialIndex?: SpatialIndex | null
+): TileResolution {
+  // TODO: Support increased tile size and resolution for dynamic H3 spatial indexes.
+  if (spatialIndex !== SpatialIndex.H3) {
+    return REDUCED_QUERIES_TILE_RESOLUTION;
+  }
+
+  return DEFAULT_TILE_RESOLUTION;
+}
+
+function scaleAggregationResLevel(
+  aggregationResLevel: number,
+  tileResolution: number
+): number | undefined {
+  if (typeof aggregationResLevel !== 'number') return;
+  return aggregationResLevel - Math.log2(0.5 / tileResolution);
+}
+
+function getColumnNameFromGeoColumn(geoColumn: string | null | undefined) {
+  if (!geoColumn) {
+    return geoColumn;
+  }
+  const parts = geoColumn.split(':');
+  return parts.length === 1 ? parts[0] : parts.length === 2 ? parts[1] : null;
+}
+
+function getSpatialIndexFromGeoColumn(geoColumn: string) {
+  const spatialIndexToSearch = geoColumn.split(':')[0];
+
+  for (const index of Object.values(SpatialIndex)) {
+    if (SpatialIndexColumn[index].includes(spatialIndexToSearch)) {
+      return index;
+    }
+  }
+  return null;
```

Comment:
i know this code is old, but it it's overly complicated and makes no sense

what it does is simply
```const x = geoColumn.split(':')[0];
if (x === SpatialIndex.H3 || x === SpatialIndex.Quadbin) return spatialIndexToSearch
return null
```